classes/common/common_base.php

```diff
@@ -252,6 +252,40 @@ public function get_config_form_elements(\MoodleQuickForm $mform, array $fields)
     protected function log($action, $info='', $logtype = logger::TYPE_INFO, $trace='') {
         logger::get_instance()->set_element($this->get_name());
         logger::get_instance()->add_to_log($logtype, $action, $info, $trace);
+
+        if ($logtype == logger::TYPE_ERROR) {
+            $this->notify($action, $info);
+        }
+    }
+
+    /**
+     * Notify of failure
+     *
+     * @param string $subject
+     * @param string $msg
+     * @return bool
+     */
+    protected function notify($action, $msg) {
+        if (empty($this->settings['notifymailto'])) {
+            return;
+        }
+
+        $notifyemails = explode(',', $this->settings['notifymailto']);
+
+        $sitename = get_site();
+        $sitename = format_string($sitename->fullname);
+        $subject = "$sitename $action error";
+
+        // Send emails.
+        $supportuser = \core_user::get_support_user();
+        $sent = true;
+        foreach ($notifyemails as $emailaddress) {
+            $userto = \totara_core\totara_user::get_external_user(trim($emailaddress));
```

Comment:
This won't work in moodle, and given that the CI check is green this means this function isn't covered by unit tests either or this should have exploded

Reply:
Test added